From 0813c8be4a287711e62ee514a910bc6a6971a1a6 Mon Sep 17 00:00:00 2001
From: topjohnwu <topjohnwu@google.com>
Date: Wed, 29 Jun 2022 20:09:12 -0700
Subject: [PATCH 3/4] Set llvm configs when building lld

---
 src/bootstrap/native.rs | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/bootstrap/native.rs b/src/bootstrap/native.rs
index b196b55c5e7..e4635b71b25 100644
--- a/src/bootstrap/native.rs
+++ b/src/bootstrap/native.rs
@@ -810,6 +810,17 @@ fn run(self, builder: &Builder<'_>) -> PathBuf {
             .define("LLVM_CONFIG_PATH", llvm_config_shim)
             .define("LLVM_INCLUDE_TESTS", "OFF");
 
+        if builder.config.llvm_thin_lto {
+            cfg.define("LLVM_ENABLE_LTO", "Thin");
+            if !target.contains("apple") {
+                cfg.define("LLVM_ENABLE_LLD", "ON");
+            }
+        }
+
+        if let Some(ref linker) = builder.config.llvm_use_linker {
+            cfg.define("LLVM_USE_LINKER", linker);
+        }
+
         if builder.config.llvm_allow_old_toolchain {
             cfg.define("LLVM_TEMPORARILY_ALLOW_OLD_TOOLCHAIN", "YES");
         }
-- 
2.36.1

