--- a/llvm/tools/llvm-config/llvm-config.cpp
+++ b/llvm/tools/llvm-config/llvm-config.cpp
@@ -424,8 +424,7 @@ int main(int argc, char **argv) {
   const bool BuiltSharedLibs = !!LLVM_ENABLE_SHARED;

   bool DyLibExists = false;
-  const std::string DyLibName =
-      (SharedPrefix + "LLVM-" + SharedVersionedExt).str();
+  const std::string DyLibName = (SharedPrefix + "LLVM." + SharedExt).str();

   // If LLVM_LINK_DYLIB is ON, the single shared library will be returned
   // for "--libs", etc, if they exist. This behaviour can be overridden with
