--- a/src/bootstrap/tool.rs
+++ b/src/bootstrap/tool.rs
@@ -617,7 +617,7 @@ fn run(self, builder: &Builder<'_>) -> PathBuf {
                 path: "src/tools/cargo",
                 is_optional_tool: false,
                 source_type: SourceType::Submodule,
-                extra_features: Vec::new(),
+                extra_features: vec!["vendored-openssl".to_string()],
             })
             .expect("expected to build -- essential tool");
