diff --git a/src/bootstrap/tool.rs b/src/bootstrap/tool.rs
index e6d27757..6aa7d06e 100644
--- a/src/bootstrap/tool.rs
+++ b/src/bootstrap/tool.rs
@@ -554,7 +554,7 @@ fn run(self, builder: &Builder<'_>) -> PathBuf {
                 path: "src/tools/cargo",
                 is_optional_tool: false,
                 source_type: SourceType::Submodule,
-                extra_features: Vec::new(),
+                extra_features: vec!["vendored-openssl".to_string()],
                 allow_features: "",
             })
             .expect("expected to build -- essential tool");
